pom.xml

```diff
@@ -247,6 +247,11 @@
             <artifactId>spring-core</artifactId>
             <version>${spring.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
+            <version>${spring-boot.version}</version>
+        </dependency>
```

Comment:
I found we have a tons of dependencies on Spring packages. Actually I imagine that this shiro-casbin middleware can be used to replace Shiro with Casbin in all cases, including Spring, SpringBoot or other places Shiro is used (something I know like Vert.x, RuoYi, JFinal, etc.)

So is it a good practice to include Spring in the source code? Does it mean that we can only replace Shiro in a Spring environment?

Reply:
I think it's related to requirements. Here, I use spring's AOP implementation's custom annotation to intercept the interface.